plugins/processors/dcos_metadata/dcos_metadata.go

```diff
@@ -0,0 +1,273 @@
+package dcos_metadata
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+
+	"github.com/mesos/mesos-go/api/v1/lib"
+	"github.com/mesos/mesos-go/api/v1/lib/agent"
+	"github.com/mesos/mesos-go/api/v1/lib/agent/calls"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli/httpagent"
+)
+
+type DCOSMetadata struct {
+	MesosAgentUrl string
+	Timeout       internal.Duration
+	RateLimit     internal.Duration
+	containers    map[string]containerInfo
+	mu            sync.Mutex
+	once          Once
+}
+
+// containerInfo is a tuple of metadata which we use to map a container ID to
+// information about the task, executor and framework.
+type containerInfo struct {
+	containerID   string
+	taskName      string
+	executorName  string
+	frameworkName string
+	taskLabels    map[string]string
+}
+
+const sampleConfig = `
+## The URL of the local mesos agent
+mesos_agent_url = "http://$NODE_PRIVATE_IP:5051"
```

Comment:
Should we use `localhost` here instead of `$NODE_PRIVATE_IP`?

Reply:
The mesos agent doesn't run on localhost. It runs on whatever is set for $LIBPROCESS_IP, which is usually the external facing interface of the host.

Reply:
We also use `$NODE_PRIVATE_IP` in the dcos_containers plugin so I was sticking with that.

Reply:
This jumped out at me because the sample config in the `README.md` uses `localhost`. (Same story for `dcos_containers`, I just realized: `localhost` in `README.md`, `$NODE_PRIVATE_IP` in the plugin's sample config.) Those should probably be consistent. And for sample config, something literal like `localhost` is probably more helpful than an undefined env var.

This is very much a nit, so I'm fine with leaving it unchanged, or using `$NODE_PRIVATE_IP` everywhere.

Reply:
I made them consistent